src/backend/access/transam/xlog.c

```diff
@@ -181,6 +183,16 @@ const struct config_enum_entry recovery_target_action_options[] = {
 	{NULL, 0, false}
 };
 
+
+typedef struct LastWrittenLsnCacheEntry
+{
+	BufferTag	key;
+	XLogRecPtr	lsn;
+	/* L2-List for LRU replacement algorithm */
+	struct LastWrittenLsnCacheEntry* next;
+	struct LastWrittenLsnCacheEntry* prev;
```

Comment:
Let's use the doubly-linked list macros from `ilist.h`. So this would become:
```
    dlist_node lru_node;
```

Reply:
I haven't heard the term "L2-List" before. It means a doubly-linked list, right?